app/models/vm_or_template/operations/snapshot.rb

```diff
@@ -150,6 +150,18 @@ def remove_all_snapshots
     raw_remove_all_snapshots
   end
 
+  def remove_all_snapshots_queue(task_id = nil)
+    MiqQueue.put_unless_exists(
```

Comment:
We're trying to get away from `put_unless_exists` plus I don't think you want/need this in this case

Reply:
What would you use instead ? `put_or_update` ?

Reply:
Just `.put` or even better `MiqTask.generic_action_with_callback` handles this for you